packages/mdc-segmented-button/segmented-button/foundation.ts

```diff
@@ -40,27 +40,38 @@ export class MDCSegmentedButtonFoundation extends MDCFoundation<MDCSegmentedButt
     super({...MDCSegmentedButtonFoundation.defaultAdapter, ...adapter});
   }
 
-  selectSegment(_indexOrSegmentId: number | string) {
-    return;
+  selectSegment(indexOrSegmentId: number | string) {
+    this.adapter.selectSegment(indexOrSegmentId);
   }
 
-  unselectSegment(_indexOrSegmentId: number | string) {
-    return;
+  unselectSegment(indexOrSegmentId: number | string) {
+    this.adapter.unselectSegment(indexOrSegmentId);
   }
 
   getSelectedSegments(): readonly SegmentDetail[] {
-    return [];
+    return this.adapter.getSegments().filter(segmentDetail => segmentDetail.selected);
   }
 
-  isSegmentSelected(_indexOrSegmentId: number | string): boolean {
-    return false;
+  isSegmentSelected(indexOrSegmentId: number | string): boolean {
+    let segment = this.adapter.getSegments().filter(segmentDetail => segmentDetail.index === indexOrSegmentId || segmentDetail.segmentId === indexOrSegmentId);
+    return segment.length > 0 && segment[0].selected;
   }
 
   isSingleSelect(): boolean {
-    return false;
+    return this.adapter.hasClass('mdc-segmented-button--single-select');
   }
 
-  handleSelected(_detail: SegmentDetail) {
-    return;
+  handleSelected(detail: SegmentDetail) {
+    if (this.isSingleSelect()) {
+      let selectedSegments = this.getSelectedSegments();
+      if (detail.selected) {
```

Comment:
can you pull this out into its own private method to prevent nested conditionals?

Reply:
If we have the child segment keep itself from being unselected then I can delete the `if (detail.selected)` logic. If we leave it up to the parent to take care of this, I'll be sure to pull this logic into a helper method!